src/base1/cockpit.js

```diff
@@ -1446,24 +1446,29 @@ function factory() {
     };
 
     cockpit.format_number = function format_number(number) {
-        /* TODO: Make the decimal separator translatable */
-
         /* We show 3 digits of precison but avoid scientific notation.
          * We also show integers without digits after the comma.
+         *
+         * We want to localise the decimal place, but we never want to
+         * show thousands separators (to avoid ambiguity).  For this
+         * reason, for integers and large enough numbers, we use
+         * non-localised conversions (and in both cases, show no
+         * fractional part).
          */
 
         if (!number && number !== 0)
             return "";
         else if (number % 1 === 0)
             return number.toString();
         else if (number > 0 && number <= 0.001)
-            return "0.001";
+            return (0.001).toLocaleString(cockpit.language);
         else if (number < 0 && number >= -0.001)
-            return "-0.001";
+            return (-0.001).toLocaleString(cockpit.language);
         else if (number > 999 || number < -999)
             return number.toFixed(0);
         else
-            return number.toPrecision(3);
+            return number.toLocaleString(cockpit.language, {maximumSignificantDigits: 3,
+                minimumSignificantDigits: 3});
```

Comment:
We usually indent option dictionaries like this:

```javascript
return number.toLocaleString(cockpit.language, {
    maximumSignificantDigits: 3,
    minimumSignificantDigits: 3
});
```

Or, if they fit on one line, like this:
return number.toLocaleString(cockpit.language, { maximumSignificantDigits: 3, minimumSignificantDigits: 3 });
```

I realize we're not very consistent about this in `cockpit.js`, though.